packages/web-component/src/utils/utils.ts

```diff
@@ -60,60 +15,56 @@ export function zip(arrays): Array<any[]> {
+ * @param {string} - the path to the readalong file
  */
-export function parseTEI(path: string): Page[] {
-  let xmlDocument =  getXML(path)
+export async function parseRAS(path: string): Promise<Array<Page>> {
+  let response = await fetch(path);
+  if (!response.ok) {
+    console.log(`fetch(${path}) failed with status ${response.status}`);
+    return [];
+  }
+  let xmlDocument = await response.text();
   let parser = new DOMParser();
-  let xml_text = parser.parseFromString(xmlDocument, "text/xml")
-  let pages = getNodeByXpath('.//div[@type="page"]', xml_text)
-  let parsed_pages = pages.map((p: Element) => {
-    let id = p.id;
-    let img_xpath = `.//div[@id='${id}']/graphic/@url`
-    let img = getNodeByXpath(img_xpath, xml_text)
-    let p_xpath = `.//div[@id='${id}']/p`
-    let paragraphs = getNodeByXpath(p_xpath, xml_text)
-    let parsed_page = { id: id, paragraphs: paragraphs }
-    if (img.length > 0) {
-      parsed_page['img'] = img[0].nodeValue;
+  let xml = parser.parseFromString(xmlDocument, "text/xml");
+  let parsed_pages = Array.from(xml.querySelectorAll("div[type=page]")).map((page) => {
+    let img = page.querySelector("graphic[url]");
+    let paragraphs = page.querySelectorAll("p");
+    let parsed_page = { id: page.getAttribute("id"), paragraphs: Array.from(paragraphs) }
+    if (img !== null) {
+      parsed_page['img'] = img.getAttribute("url");
     }
-    if(p.attributes)parsed_page["attributes"]=p.attributes;
-    return parsed_page
+    if (page.attributes)
+      parsed_page["attributes"] = page.attributes;
+    return parsed_page;
   });
-  return parsed_pages
+  return parsed_pages;
 }
 
 
-
 /**
- * Return useful data from SMIL xml file
- * @param {string} - the path to the SMIL file
+ * Extract alignment data from parsed text
  */
-export function parseSMIL(path: string): Alignment {
-  let xmlDocument = getXML(path)
-  let parser = new DOMParser();
-  let xml_text = parser.parseFromString(xmlDocument, "text/xml")
-  let text = getNodeByXpath('/i:smil/i:body/i:par/i:text/@src', xml_text).map(x => {
-    let split = x['value'].split('#');
-    return split[split.length - 1]
-  }
-  )
-  let audio_begin = getNodeByXpath('/i:smil/i:body/i:par/i:audio/@clipBegin', xml_text).map(x => x['value'] * 1000)
-  let audio_end = getNodeByXpath('/i:smil/i:body/i:par/i:audio/@clipEnd', xml_text).map(x => x['value'] * 1000)
-  let audio_duration = []
-  for (var i = 0; i < audio_begin.length; i++) {
-    let duration = audio_end[i] - audio_begin[i]
-    audio_duration.push(duration)
-  }
-  let audio = zip([audio_begin, audio_duration])
-  let result = {}
-  for (var i = 0; i < text.length; i++) {
-    result[text[i]] = audio[i]
+export function extractAlignment(parsed_text: Array<Page>): Alignment {
+  let alignment = {};
+  for (const page of parsed_text) {
+    for (const p of page.paragraphs) {
+      const words = Array.from(p.getElementsByTagName("w"));
+      for (const w of words) {
```

Comment:
Does this code assume that the structure is of fixed depth, with `<p>` elements always immediately under the page `<div>` elements, w under p, and no variations? Or will this search for any p elements under the page, any w elements under the paragraph, etc? It seems to me the original code was quite flexible, allowing for structures that were not necessarily exactly what Studio produces.

I don't know if we have use cases for such structures at the moment, but I could imagine wanting to use spans for changing colour or other font aspects for several words/paragraphs in a row. I don't have a concrete use case for such formatting, but if we decide we stop supporting it, I'd like to make sure we're intentionally making that decision, not accidentally.

Reply:
I guess that same question/comment applies to the DTD as well.

I think I like making the structure stricter, because it will make things simpler. But again only if we're doing so deliberately.

Reply:
Good question, but yes, `getElementsByTagName` will find `<w>` at any depth.  It is like a CSS selector (in fact we could use `querySelectorAll` to be more explicit about this)

Reply:
Ah, OK, then all is good.